bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,271 @@
+                        end_column=m.end(0) + 1,
+                        diffs=diff)
+
+            # check for control start tags
+            for m in control_start_regex.finditer(line):
+                # build the label which is expected at the end of this block
+                end_label = "{{#{content}#}}".format(
+                    content=m.group("content"))
+                control_stack.append((end_label, line_number))
+
+                # check for spacing issues with this tag
+                if not is_spacing_valid(m.group("content"), control_spacing):
+                    yield self.handle_control_spacing_issue(
+                        file, filename, line, line_number, control_spacing, m)
+
+            # check for control end tags
+            for m in control_end_regex.finditer(line):
+                label = m.group("label")
+                try:
+                    (expected_label, start_in_line) = control_stack.pop()
+                except IndexError:
+                    # and end tag without an opening tag
+                    yield Result.from_values(
+                        origin=self,
+                        message="Control end tag has no corresponding start",
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
+                        end_line=line_number + 1,
+                        end_column=m.end(0) + 1,
+                        severity=RESULT_SEVERITY.MAJOR)
+                    continue
+
+                # check for spacing issues with this tag
+                if not is_spacing_valid(m.group("content"), control_spacing):
+                    yield self.handle_control_spacing_issue(
+                        file, filename, line, line_number, control_spacing, m)
+
+                # yield results for incorrect or missing end labels
+                if label is None and line_number != start_in_line:
+                    diff = generate_label_diff(
+                        file, filename, line, line_number, m, expected_label)
+                    yield Result.from_values(
+                        origin=self,
+                        message="Untagged control end tag",
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
+                        end_line=line_number,
+                        end_column=m.end(0) + 1,
+                        diffs=diff)
+                elif label != expected_label and line_number != start_in_line:
+                    diff = generate_label_diff(
+                        file, filename, line, line_number, m, expected_label)
+                    yield Result.from_values(
+                        origin=self,
+                        message="Tag does not match expected label",
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
```

Comment:
This one should have the columns of the label rather than the tag.
So, `m.start("label")`/`m.end("label")` or `m.start(3)`, ..